File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size() - 1;
        int width = j;
        int ans = INT_MIN;
        while(i < j){
            int m1 = min(height[i],height[j]);
            ans = max(ans, m1*width--);
            if(height[i]<height[j])
                i++;
            else
                j--;
        }
        return ans;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int i = 0, j = n-1;
        int ans = 0;
        while(i < j){
            int mini = Math.min(height[i],height[j]);
            ans = Math.max(ans, (j-i) * mini);
            if(height[i] < height[j])
                i++;
            else
                j--;
        }
        return ans;
    }
}